components/ApplicationView/ApplicationView.jsx

```diff
@@ -11,22 +12,48 @@ const ApplicationView = ({ applicationId }) => {
   const [data, setData] = useState();
   const [status, setStatus] = useState();
   const [error, setError] = useState(false);
+  const [validationRecap, setValidationRecap] = useState();
+  const { register, watch, reset } = useForm({ defaultValues: {} });
+  const watcher = watch({ nest: true });
+  const validations = JSON.stringify(watcher);
   const fetchData = useCallback(async applicationId => {
     if (!applicationId) {
       return null;
     }
     setError(false);
     try {
-      const { data } = await axios.get(`/api/applications/${applicationId}`);
-      setData(data.application);
+      const { application, validations } = await fetchApplication(
+        applicationId
+      );
+      setData(application);
+      validations && reset(JSON.parse(validations));
     } catch (e) {
       setError(e.response.data);
     }
   }, []);
+  const saveValidation = useCallback(async validations => {
+    try {
+      await patchApplication(applicationId, { validations });
+    } catch {
+      fetchData(applicationId);
+    }
+  });
+  const getValidationRecap = useCallback(watcher =>
+    Object.entries(watcher).reduce(
+      (acc, [key, value]) => ({
+        ...acc,
+        [key]: Object.values(value).every(Boolean)
+      }),
+      {}
+    )
+  );
   useEffect(() => {
     fetchData(applicationId);
   }, [applicationId]);
-
+  useEffect(() => {
+    validations !== '{}' && saveValidation(validations);
```

Comment:
I think you may need to cater for it being an empty string here, which it what it will currently be initially. That or explicitly insert `{}` when the application is submitted.

Reply:
`validation` is `JSON.stringify` of a watch, and it's initialised to `{}`
```
  const watcher = watch({ nest: true });
  const validations = JSON.stringify(watcher);
```